js_methods.rb

```diff
@@ -0,0 +1,253 @@
+# frozen_string_literal: true
+
+# rubocop:disable Style/StringLiterals, Style/AndOr, CyclomaticComplexity, PerceivedComplexity, ModuleLength
+
+module Enumerable
+  def my_each
+    return to_enum(__method__) unless block_given?
+
+    i = 0
+    while i < length
+      yield to_a[i] if block_given?
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(__method__) unless block_given?
+
+    i = 0
+    while i < length
+      yield to_a[i], i if block_given?
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(__method__) unless block_given?
+
+    temp_arr = []
+    my_each do |e|
+      temp = yield e
+      temp_arr << e if temp
+    end
+    temp_arr
+  end
+
+  def my_all?(*vars)
+    ans = true
+    if block_given?
+      my_each { |e| ans = false unless yield e }
+    elsif !vars.empty?
+      if vars[0].is_a?(Class) # 2/4
+        my_each { |e| ans = false unless e.is_a?(vars[0]) }
+      elsif vars[0].is_a?(Regexp) # 3/4
+        my_each { |e| ans = false unless e =~ vars[0] }
+      else
+        my_each { |e| ans = false unless e == vars[0] } # 4/4
+      end
+    else
+      my_each { |e| ans = false if e == false || e.nil? } # 1/4
+    end
+    ans
+  end
+
+  def my_any?(*vars)
+    !my_none?(*vars)
+  end
+
+  def my_none?(*vars)
+    ans = true
+    if block_given?
+      my_each { |e| ans = false if yield e }
+    elsif !vars.empty?
+      if vars[0].is_a?(Class) # 2/4
+        my_each { |e| ans = false if e.is_a?(vars[0]) }
+      elsif vars[0].is_a?(Regexp) # 3/4
+        my_each { |e| ans = false if e =~ vars[0] }
+      else
+        my_each { |e| ans = false if e == vars[0] } # 4/4
+      end
+    else
+      my_each { |e| ans = false if e != false || e.nil? } # 1/4
+    end
+    ans
+  end
+
+  def my_count(*chr)
+    # why can't use *char directly???
+    cnt = 0
+    char = *chr
+    if block_given?
+      my_each { |e| cnt += 1 if yield e }
+      cnt
+    elsif !char.empty?
+      my_each { |e| cnt += 1 if e == char[0] }
+      cnt
+    else
+      length
+    end
+  end
+
+  def my_map
+    arr = []
+    if block_given?
+      my_each do |e|
+        elm = yield e
+        arr << elm
+      end
+      arr
+    else
+      to_enum(__method__)
+    end
+  end
+
+  def my_inject(*init)
+    init, sym = *init
+    init.is_a?(Integer) ? acc = init : sym = init
+    if block_given? || !sym.empty?
+      my_each do |e|
+        acc = e and next if acc.nil? # to not sum 2times 1st value
+        acc = if block_given? # because if and else asign to the same variable
+                yield acc, e
+              else
+                acc.send(sym, e)
+              end
+      end
+    else
+      to_enum(__method__)
+    end
+    acc
+  end
```

Comment:
On `line 109`, that call to my_each throws an error when a range is passed. Did you make your `my_each` to work for when a range is passed?

I'd advise you to test with every enumerable after fixing it.

Reply:
done